src/main/java/com/igot/cb/designation/service/impl/DesignationServiceImpl.java

```diff
@@ -112,6 +122,95 @@ public void loadDesignationFromExcel(MultipartFile file) {
         });
     log.info("DesignationServiceImpl::loadDesignationFromExcel::created the designations");
   }
+  @Override
+  public ApiResponse createDesignation(JsonNode request){
+    ApiResponse response = ProjectUtil.createDefaultResponse(Constants.API_DESIGNATION_CREATE);
+    payloadValidation.validatePayload(Constants.DESIGNATION_CREATE_PAYLOAD_VALIDATION,request);
+    String name =  request.get(Constants.NAME).asText();
+    String ref_Id =  request.get(Constants.REF_ID).asText();
+    String ref_Type = request.get(Constants.REF_TYPE).asText();
+    Optional<DesignationEntity> designationEntity = designationRepository.findByIdAndIsActive(ref_Id,Boolean.TRUE);
+    if(designationEntity.isPresent()) {
+      DesignationEntity designation = designationEntity.get();
+      if (designation.getIsActive()) {
+      ApiResponse readResponse = readDesignation(ref_Id);
+        readResponse.setResponseCode(HttpStatus.NOT_FOUND);
+        if (readResponse == null) {
+          response.getParams().setErr("Failed to validate sector exists or not.");
+          response.setResponseCode(HttpStatus.INTERNAL_SERVER_ERROR);
+          response.getParams().setStatus(Constants.FAILED);
+        } else if (HttpStatus.NOT_FOUND.equals(readResponse.getResponseCode())) {
+          Map<String, Object> reqBody = new HashMap<>();
+          reqBody.put(Constants.CODE, ref_Id);
+          reqBody.put(Constants.REF_ID, ref_Id);
+          reqBody.put(Constants.REF_TYPE, ref_Type);
+          reqBody.put(Constants.NAME, name);
+          Map<String, Object> parentObj = new HashMap<>();
+          parentObj.put(Constants.IDENTIFIER,
+                  cbProperties.getOdcsFrameworkName() + "_" + cbProperties.getOdcsCategoryName());
```

Comment:
both of this property looks like common one... instead these should be named with odcsDesignationFramework and odcsDesignationCategory